word-search.py

```diff
@@ -334,8 +335,29 @@ def scoreWord(word):
         score = len(word) - 2
     return score
 
-def printWordList():
-    # prints all possible words to be found at game end
+def print_word_list(gridWordList, foundWords):
+    # Prints all possible words to be found at game end
+    # Reveals player's answers and possible answers
+    # Group words by their lengths from gridWordList
+    word_groups = defaultdict(list)
+
+    # Sort the words within each group
+    for word in gridWordList:
+        word_groups[len(word)].append(word)
+
+    # Found words first, then not found, then list alphabetically
+    for length in word_groups:
```

Comment:
tested update, currently found words aren't printed first. I think this is okay, but it would be better if found words are mentioned first ?
![image](https://github.com/user-attachments/assets/e35c9e43-38a7-4353-be63-8df1ef5f8e92)

Reply:
Thank you for testing it. This is noted. I'll make some changes :))

Reply:
Thanks for making changes. Tested, looks good, output is categorized by word length and displayed found words first, followed by not found. 
![image](https://github.com/user-attachments/assets/38d1df6c-4982-488d-9f4f-790b4d36b999)